### PR TITLE
PRINT does not deserialize a DELIMITED topic with one integer column #2639

### PR DESCRIPTION
**Pull request by spena**
Fixes https://github.com/confluentinc/ksql/issues/2639

### Description 
This fixes the issue when a topic has one single value with one column.
```
# echo "1000" | kafkacat -b localhost:9092 -t t1 -P

ksql> print 't1' from BEGINNING;
com.fasterxml.jackson.databind.node.IntNode cannot be cast to com.fasterxml.jackson.databind.node.ObjectNode
```

The issue is that `PRINT` was selecting a JSON format to display the above value (which is a valid JSON value), however, the `PRINT` format was expecting to use an `ObjectNode` which expects a `key:value` format. 

The fix is easy. Just to prevent using JSON if the value is not an `ObjectNode`. It will fall back to the STRING format instead.

### Testing done 
Add a test to `TopicStreamTest`
Manually verified in KSQL CLI
```
ksql> print 't1' from BEGINNING;
Format:STRING
4/12/19 9:55:55 AM CDT , NULL , 1000
```

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/syntax-reference.rst [hunk 1]
```diff
@@ -445,12 +445,18 @@ The WITH clause for the result supports the following properties:
...
+|                         | For KSQL 5.2 and earlier, if the property is not set, the value of the ``ksql.sink.partitions``      |
+|                         | property, which defaults to four partitions, will be used. The ``ksql.sink.partitions`` property can |
+|                         | be set in the properties file the KSQL server is started with, or by using the ``SET`` statement.    |
 +-------------------------+------------------------------------------------------------------------------------------------------+
 | REPLICAS                | The replication factor for the topic. If this property is not set, then the number of                |
-|                         | replicas of the input stream or table will be used.                                                  |
+|                         | replicas of the input stream or table will be used. In join queries, the property values are taken   |
+|                         | from the left-side stream or table.                                                                  |
+|                         | For KSQL 5.2 and earlier, if the REPLICAS is not set, the value of the ``ksql.sink.replicas``        |
```
Comment (JimGalasyn) [inline docs/developer-guide/syntax-reference.rst]:
```suggestion
|                         | For KSQL 5.2 and earlier, if REPLICAS is not set, the value of the ``ksql.sink.replicas``            |
```

File: docs/developer-guide/syntax-reference.rst [hunk 2]
```diff
@@ -526,12 +532,18 @@ The WITH clause supports the following properties:
...
```
Comment (JimGalasyn) [inline docs/developer-guide/syntax-reference.rst]:
```suggestion
|                         | For KSQL 5.2 and earlier, if REPLICAS is not set, the value of the ``ksql.sink.replicas``            |
```

File: docs/installation/server-config/config-reference.rst [hunk 3]
```diff
@@ -158,21 +158,33 @@ By default, the service ID of KSQL servers is ``default_``. The service ID is al
 the prefix for the internal topics created by KSQL. Using the default value ``ksql.service.id``, the KSQL internal topics
 will be prefixed as ``_confluent-ksql-default_`` (e.g. ``_command_topic`` becomes ``_confluent-ksql-default__command_topic``).
 
+.. _ksql-internal-topic-replicas:
+
+--------------------
```
Comment (JimGalasyn) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
----------------------------
```

File: docs/installation/server-config/config-reference.rst [hunk 4]
```diff
@@ -158,21 +158,33 @@ By default, the service ID of KSQL servers is ``default_``. The service ID is al
...
+ksql.internal.topic.replicas
+--------------------
```
Comment (JimGalasyn) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
----------------------------
```

File: docs/installation/server-config/config-reference.rst [hunk 5]
```diff
@@ -158,21 +158,33 @@ By default, the service ID of KSQL servers is ``default_``. The service ID is al
...
+
+The number of replicas for the internal topics created by KSQL Server. The default is one.
+This config parameter works in KSQL 5.3 and later.
+Note that replicas for record processing log topic should be configrued separately. For more info refer to :ref:`KSQL Processing Log <ksql_processing_log>`.
```
Comment (JimGalasyn) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
Replicas for the record processing log topic should be configured separately. For more information, see :ref:`KSQL Processing Log <ksql_processing_log>`.
```

File: docs/installation/server-config/config-reference.rst [hunk 6]
```diff
@@ -158,21 +158,33 @@ By default, the service ID of KSQL servers is ``default_``. The service ID is al
...
+
 .. _ksql-sink-partitions:
 
 --------------------
```
Comment (JimGalasyn) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
---------------------------------
```

File: docs/installation/server-config/config-reference.rst [hunk 7]
```diff
@@ -158,21 +158,33 @@ By default, the service ID of KSQL servers is ``default_``. The service ID is al
...
-ksql.sink.partitions
+ksql.sink.partitions (Deprecated)
 --------------------
```
Comment (JimGalasyn) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
---------------------------------
```

File: docs/installation/server-config/config-reference.rst [hunk 8]
```diff
@@ -158,21 +158,33 @@ By default, the service ID of KSQL servers is ``default_``. The service ID is al
...
 
 The default number of partitions for the topics created by KSQL. The default is four.
+This property has been deprecated since 5.3 release. For more info see the WITH clause properties in :ref:`CREATE STREAM AS SELECT <create-stream-as-select>` and :ref:`CREATE TABLE AS SELECT <create-table-as-select>`.
```
Comment (JimGalasyn) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
Starting in version 5.3, this property is deprecated. For more information, see the WITH clause properties in :ref:`CREATE STREAM AS SELECT <create-stream-as-select>` and :ref:`CREATE TABLE AS SELECT <create-table-as-select>`.
```

File: docs/installation/server-config/config-reference.rst [hunk 9]
```diff
@@ -158,21 +158,33 @@ By default, the service ID of KSQL servers is ``default_``. The service ID is al
...
 
 .. _ksql-sink-replicas:
 
 ------------------
```
Comment (JimGalasyn) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
-------------------------------
```

File: docs/installation/server-config/config-reference.rst [hunk 10]
```diff
@@ -158,21 +158,33 @@ By default, the service ID of KSQL servers is ``default_``. The service ID is al
...
-ksql.sink.replicas
+ksql.sink.replicas (Deprecated)
 ------------------
```
Comment (JimGalasyn) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
-------------------------------
```

File: docs/installation/server-config/config-reference.rst [hunk 11]
```diff
@@ -158,21 +158,33 @@ By default, the service ID of KSQL servers is ``default_``. The service ID is al
...
 
 The default number of replicas for the topics created by KSQL. The default is one.
+This property has been deprecated since 5.3 release. For more info see the WITH clause properties in :ref:`CREATE STREAM AS SELECT <create-stream-as-select>` and :ref:`CREATE TABLE AS SELECT <create-table-as-select>`.
```
Comment (JimGalasyn) [inline docs/installation/server-config/config-reference.rst]:
```suggestion
Starting in version 5.3, this property is deprecated. For more information, see the WITH clause properties in :ref:`CREATE STREAM AS SELECT <create-stream-as-select>` and :ref:`CREATE TABLE AS SELECT <create-table-as-select>`.
```

**Comment by spena** [comment]
I didn't notice, this patch is not valid for 5.2.x. I don't know how it ended up having so many changes. It's supposed to be just 1 file. I will upload another PR with the correction.

**Pull request closed without merging**